esp-hal/src/spi/master.rs

```diff
@@ -1565,96 +1535,135 @@ pub mod dma {
         }
     }
 
-    /// A DMA-capable SPI bus that handles full-duplex transfers.
+    #[derive(Default)]
+    enum State<'d, T, C, D, M>
+    where
+        T: InstanceDma,
+        C: DmaChannel,
+        C::P: SpiPeripheral,
+        D: DuplexMode,
+        M: Mode,
+    {
+        Idle(SpiDma<'d, T, C, D, M>, DmaTxBuf, DmaRxBuf),
+        Reading(SpiDmaTransfer<'d, T, C, D, M, DmaRxBuf>, DmaTxBuf),
+        Writing(SpiDmaTransfer<'d, T, C, D, M, DmaTxBuf>, DmaRxBuf),
+        Transferring(SpiDmaTransfer<'d, T, C, D, M, (DmaTxBuf, DmaRxBuf)>),
+        #[default]
+        InUse,
```

Comment:
This needs an explanation (or a rename to `TemporarilyRemoved` or something), or we can use the [replace_with](https://crates.io/crates/replace_with) crate to remove it.